### ISSUE TITLE
Make @TestTarget accept a Kotlin field

### ISSUE THREAD

**Issue by pkubowicz** [DiUS/pact-jvm#871]
Writing a provider test in Kotlin with
```
    @TestTarget
    val target: Target = HttpTarget(port = 9052)
```
causes an exception:
```
How did getFields return a field we couldn't access?
java.lang.RuntimeException: How did getFields return a field we couldn't access?
	at org.junit.runners.model.TestClass.getAnnotatedFieldValues(TestClass.java:235)
	at au.com.dius.pact.provider.junit.InteractionRunner.lookupTarget(InteractionRunner.kt:223)
	at au.com.dius.pact.provider.junit.InteractionRunner.interactionBlock(InteractionRunner.kt:199)
	at au.com.dius.pact.provider.junit.InteractionRunner.run(InteractionRunner.kt:144)
	at au.com.dius.pact.provider.junit.PactRunner.runChild(PactRunner.kt:137)
	at au.com.dius.pact.provider.junit.PactRunner.runChild(PactRunner.kt:51)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
Caused by: java.lang.IllegalAccessException: class org.junit.runners.model.FrameworkField cannot access a member of class com.example.ServerContractTest with modifiers "private final"
	at java.base/jdk.internal.reflect.Reflection.newIllegalAccessException(Reflection.java:355)
	at java.base/java.lang.reflect.AccessibleObject.checkAccess(AccessibleObject.java:639)
	at java.base/java.lang.reflect.Field.checkAccess(Field.java:1075)
	at java.base/java.lang.reflect.Field.get(Field.java:416)
	at org.junit.runners.model.FrameworkField.get(FrameworkField.java:73)
	at org.junit.runners.model.TestClass.getAnnotatedFieldValues(TestClass.java:230)
```

To make the test running, I need to annotate the field with `@JvmField`, which makes the Kotlin code less idiomatic. The exception is also a bit cryptic and may discourage Kotlin newcomers from using Pact (see #784)

Maybe Pact's InteractionRunner.kt could first try getAnnotatedMethodValues before running getAnnotatedFieldValues?

The whole test file can be seen in https://github.com/pkubowicz/contract-testing-samples/blob/master/pact-provider-kotlin/src/test/kotlin/com/example/ServerContractTest.kt

**Comment by uglyog** [comment]
That message is generated by the JUnit test support class. `getAnnotatedMethodValues` won't work because the annotation won't be on the accessor unless you write 

```
    val target: Target = HttpTarget(port = 9052)
        @TestTarget get
```

which I think is not any better.

**Comment by uglyog** [comment]
The solution was to first look for a getter for the field and call that, otherwise try access the field directly.

**Comment by uglyog** [comment]
This fix has been released with version 3.6.4
